### PR TITLE
Remove the SHOW_EIP_1559_UI environment variable, replace with network detection where appropriate

### LINKED ISSUE #11490: EIP-1559 - Audit Usage of SHOW_EIP_1559_UI
At present, we use the `SHOW_EIP_1559_UI` environment variable for two purposes:

1.  Whether we want the EIP-1559 components to be used in the extension (so that we could merge work into the extension without the users seeing it when we have releases before 1559)
2. Whether we want to show pieces of UI or old UI

When we have components wired up, we need to remove `SHOW_EIP_1559_UI` and instead replace that functionality with logic to check if a given transaction or flow is an EIP-1559 flow.

### PR DESCRIPTION
**Pull request by darkwing**
Fixes: https://github.com/MetaMask/metamask-extension/issues/11490

I did the best I could to keep this small, minimal.  We'll need follow up work to remove extraneous props, remove unused components.  The difficult one was `ui/pages/confirm-transaction-base/confirm-transaction-base.component.js`, so have a good look at that one.

### CODE REVIEW

**Comment by darkwing** [comment]
Test updates look good , but since I did the logic part, I think others should review.

**Review by danjm — APPROVED** [review]
This looks good to me. Let's make sure to get some QA on it before merge.

**Review by danjm — CHANGES_REQUESTED** [review]
hold on, need to re-check something

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js
Comment (danjm) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:307]: I'm confused about something. A lot of code deleted in this file seems to be necessary for EIP1559. And some that is deleted seems necessary non-EIP1559 support. For example, on develop `inlineGasControls` (either `AdvancedGasControls` or `AdvancedGasControls`) is shown on both EIP1559 networks and non-EIP1559 networks, but here it is removed completely.

Why not just replace the `process.env.SHOW_EIP_1559_UI` checks with a check for whether the current network supports eip1559?

Comment (danjm) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:307]: I think I missed this when I first reviewed because the diff of this file was hidden

Comment (darkwing) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:307]: I was under the impression that all gas editing should go through the popover.  Also, we had spoken about removing the inline controls for now, since we need to restructure the data handling.

Comment (danjm) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:307]: Okay, actually, my assessment above is not right. There are some things removed: the `inlineGasControls` functionality is removed with this PR, and so is the `primaryTotalTextOverride` and `secondaryTotalTextOverride`. We still should support these.

Comment (danjm) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:307]: I can see the difficulty in having inline gas controls here, but I think we need to continue to support their use? There removal would be a degradation of UX on non-eip1559 networks.

Comment (adonesky1) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:307]: @danjm is it a degradation if the functionality is still available but just in the popover? Plus on non-eip1559 networks the gas controls are available on the send stage? So for a non-1559 network the user could either use the gas pop over to edit gas on the confirmation screen or navigate back to the send screen and edit gas from there?

File: ui/pages/send/send-content/send-gas-row/send-gas-row.component.js
Comment (danjm) [inline ui/pages/send/send-content/send-gas-row/send-gas-row.component.js:49]: I don't think we should show the customization modal in send when on an EIP-1559 network

Comment (danjm) [reply ui/pages/send/send-content/send-gas-row/send-gas-row.component.js:49]: This code can't be hit, but it should be removed

**Comment by darkwing** [comment]
Closing in favor of Alex's PR!

**Pull request closed without merging**
